### PR TITLE
API: change default 'align' for bar and barh

### PR DESCRIPTION
**Pull request by tacaswell**
Change from 'edge' to 'center'

so

### CODE REVIEW

File: doc/users/dflt_style_changes.rst [hunk 1]
```diff
@@ -529,6 +529,39 @@ or by setting::
...
+.. plot::
+
+   import matplotlib.pyplot as plt
+   import numpy as np
+
+   fig, ((ax1, ax2), (ax3, ax4)) = plt.subplots(2, 2, figsize=(5, 5))
+
+   def demo(bar_func, bar_kwargs):
+       return bar_func([1, 2, 3], [1, 2, 3], tick_label=['a', 'b', 'c'],
+                       ,**bar_kwargs)
```
Comment (efiring) [inline doc/users/dflt_style_changes.rst]: Delete comma.

File: doc/users/dflt_style_changes.rst [hunk 2]
```diff
@@ -529,6 +529,39 @@ or by setting::
...
+
+   ax2.set_title("classic alignment")
+
+   demo(ax1.bar, {})
+   demo(ax2.bar, {'align': 'edge'})
+   demo(ax3.barh, {})
+   demo(ax4.barh, {'align': 'edge'})
+
+
+To restore the previous default explicitly pass a value for ``align='edge'`` to the
```
Comment (efiring) [inline doc/users/dflt_style_changes.rst]: "a value for" -> "the kwarg"

File: doc/users/dflt_style_changes.rst [hunk 3]
```diff
@@ -529,6 +529,39 @@ or by setting::
 
 in your :file:`matplotlibrc` file.
 
+``bar`` and ``barh``
+====================
+
+The default value of the ``align`` kwarg for both
+`~matplotlib.Axes.bar` and `~matplotlib.Axes.barh` is changed to
+``'center'`` from ``'edge'``.
```
Comment (efiring) [inline doc/users/dflt_style_changes.rst]: Nit: this would read more smoothly with the reversed order, "from center to edge".

**Comment by efiring** [comment]
Travis docbuild also caught the extraneous comma.

**Comment by tacaswell** [comment]
@efiring fixed and force-pushed

File: doc/users/dflt_style_changes.rst [hunk 4]
```diff
@@ -529,6 +529,39 @@ or by setting::
...
+`~matplotlib.Axes.bar` and `~matplotlib.Axes.barh` is changed from
+``'edge'`` to ``'center'``.
+
+
+.. plot::
+
+   import matplotlib.pyplot as plt
+   import numpy as np
+
+   fig, ((ax1, ax2), (ax3, ax4)) = plt.subplots(2, 2, figsize=(5, 5))
```
Comment (QuLogic) [inline doc/users/dflt_style_changes.rst]: Why put this before the function?

File: doc/users/dflt_style_changes.rst [hunk 5]
```diff
@@ -529,6 +529,39 @@ or by setting::
...
+
+   ax2.set_title("classic alignment")
+
+   demo(ax1.bar, {})
+   demo(ax2.bar, {'align': 'edge'})
+   demo(ax3.barh, {})
+   demo(ax4.barh, {'align': 'edge'})
+
+
+To restore the previous default explicitly pass a the kwarg
```
Comment (QuLogic) [inline doc/users/dflt_style_changes.rst]: a the?

And I would spell out the whole words.

File: lib/matplotlib/tests/test_artist.py [hunk 6]
```diff
@@ -185,7 +185,7 @@ def test_default_edges():
 
     ax1.plot(np.arange(10), np.arange(10), 'x',
              np.arange(10) + 1, np.arange(10), 'o')
-    ax2.bar(np.arange(10), np.arange(10))
+    ax2.bar(np.arange(10), np.arange(10), align='edge')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_artist.py]: It's not quite "default" edges now... Not strictly necessary to update, though.

File: doc/users/dflt_style_changes.rst (hunk 4 above)
Comment (tacaswell) [inline doc/users/dflt_style_changes.rst]: No particular reason.  A bunch of the example on this page page this ordering.

**Pull request merged**